src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractChangeNamespaceService.cs

```diff
@@ -773,40 +767,39 @@ await FixReferencesAsync(document, changeNamespaceService, addImportService, ref
         CodeCleanupOptionsProvider fallbackOptions,
         CancellationToken cancellationToken)
     {
-        using var _ = PooledHashSet<DocumentId>.GetInstance(out var linkedDocumentsToSkip);
-        var documentsToProcessBuilder = ArrayBuilder<Document>.GetInstance();
+        using var _1 = PooledHashSet<DocumentId>.GetInstance(out var linkedDocumentsToSkip);
+        using var _2 = ArrayBuilder<Document>.GetInstance(out var documentsToProcess);
 
         foreach (var id in ids)
         {
             if (linkedDocumentsToSkip.Contains(id))
-            {
                 continue;
-            }
 
             var document = solution.GetRequiredDocument(id);
             linkedDocumentsToSkip.AddRange(document.GetLinkedDocumentIds());
-            documentsToProcessBuilder.Add(document);
-
-            document = await RemoveUnnecessaryImportsWorkerAsync(
-                document,
-                CreateImports(document, names, withFormatterAnnotation: false),
-                cancellationToken).ConfigureAwait(false);
```

Comment:
this was wacky.  we were updating the documents serially, and then doing a parallel processing of it below.